libbpf-cargo/src/build.rs

```diff
@@ -111,26 +111,36 @@ fn check_clang(debug: bool, clang: &Path, skip_version_checks: bool) -> Result<(
 ///
 /// for each prog.
 fn compile_one(debug: bool, source: &Path, out: &Path, clang: &Path, options: &str) -> Result<()> {
-    let arch = match std::env::consts::ARCH {
-        "x86_64" => "x86",
-        "aarch64" => "arm64",
-        _ => std::env::consts::ARCH,
-    };
-
     if debug {
         println!("Building {}", source.display());
     }
 
     let mut cmd = Command::new(clang.as_os_str());
+    let mut arch_set = false;
+
+    // When doing cross-compile, user may set -D__TARGET_ARCH_<arch> in clang_args.
     if !options.is_empty() {
         cmd.args(options.split_whitespace());
+
+        if options.contains("-D__TARGET_ARCH_") {
+            arch_set = true;
+        }
     }
+
+    if !arch_set {
+        let arch = match std::env::consts::ARCH {
+            "x86_64" => "x86",
+            "aarch64" => "arm64",
+            _ => std::env::consts::ARCH,
+        };
+        cmd.arg(format!("-D__TARGET_ARCH_{}", arch));
+    }
```

Comment:
I think this logic could be simplified with something like:


    let mut cmd = Command::new(clang.as_os_str());

    if !options.is_empty() {
        cmd.args(options.split_whitespace());
    }

    if !options.contains("-D__TARGET_ARCH_") {
        let arch = match std::env::consts::ARCH {
            "x86_64" => "x86",
            "aarch64" => "arm64",
            _ => std::env::consts::ARCH,
        };
        cmd.arg(format!("-D__TARGET_ARCH_{}", arch));
    }


and that way we don't need the arch_set. What are your thoughts?

Reply:
I think your fix is better!
Is it possible for you to update the patch and add a "Signed-off-by" when merging the code?

Reply:
I will try to figure out how to do this in github :) What "Signed-off-by" name should I put for you? 

And if I can't figure this out, then I will just land this as-is and submit a follow-up. 

Thanks for contributing to libbpf-rs.

Reply:
> I will try to figure out how to do this in github :)  What "Signed-off-by" name should I put for you?

Thank you.
Please use follows:
     
     Signed-off-by: Seimizu Joukan <joukan.seimizu@gmail.com>